src/meta/src/hummock/manager/compaction.rs

```diff
@@ -257,6 +258,78 @@ impl HummockManager {
             } => {}
         }
     }
+
+    pub(crate) async fn calculate_vnode_partition(
+        &self,
+        compact_task: &mut CompactTask,
+        compaction_config: &CompactionConfig,
+    ) {
+        if compact_task.target_level > compact_task.base_level {
+            return;
+        }
+        if compaction_config.split_weight_by_vnode > 0 {
+            for table_id in &compact_task.existing_table_ids {
+                compact_task
+                    .table_vnode_partition
+                    .insert(*table_id, compact_task.split_weight_by_vnode);
+            }
+        } else {
+            let mut table_size_info: HashMap<u32, u64> = HashMap::default();
+            let mut existing_table_ids: HashSet<u32> = HashSet::default();
+            for input_ssts in &compact_task.input_ssts {
+                for sst in &input_ssts.table_infos {
+                    existing_table_ids.extend(sst.table_ids.iter());
+                    for table_id in &sst.table_ids {
+                        *table_size_info.entry(*table_id).or_default() +=
+                            sst.file_size / (sst.table_ids.len() as u64);
+                    }
+                }
+            }
+            compact_task
+                .existing_table_ids
+                .retain(|table_id| existing_table_ids.contains(table_id));
+
+            let hybrid_vnode_count = self.env.opts.hybrid_partition_node_count;
+            let default_partition_count = self.env.opts.partition_vnode_count;
+            // We must ensure the partition threshold large enough to avoid too many small files.
+            let less_partition_threshold = self.env.opts.hybrid_few_partition_threshold;
+            let several_partition_threshold = self.env.opts.hybrid_more_partition_threshold;
```

Comment:
nits: Let's keep the variable naming consistent with the opts naming
```
let compact_task_table_size_split_threshold_low = self.env.opts.compact_task_table_size_split_threshold_low
let compact_task_table_size_split_threshold_high = self.env.opts.compact_task_table_size_split_threshold_high
```

Reply:
Fixed.